### PR TITLE
Add framebuffer bindings to WebGL state manager.

### LINKED ISSUE #316: Add framebuffer bindings to WebGL state manager.
Using `withParameters` to recursively override framebuffers, but on exit it doesn't reset to previous, it always resets to null (the default framebuffer) - this breaks rendering badly.

We should integrate WebGL bindings (at least frame buffer bindings) into state management.

### PR DESCRIPTION
**Pull request by 1chandu**
Fixes #316

### CODE REVIEW

File: src/webgl-utils/track-context-state.js [hunk 1]
```diff
@@ -45,9 +45,10 @@ export const GL_STATE_SETTERS = {
 
   // SPECIFIC SETTERS
 
-  clearStencil: (setter, s) => setter({
-    [GL.STENCIL_CLEAR_VALUE]: s}
-  ),
+  bindFramebuffer: (setter, target, fb) => setter({
```
Comment (ibgreen) [inline src/webgl-utils/track-context-state.js]: Per my read of the article I forwarded, it should be 
```
bindFramebuffer: (setter, target, fb) => {
   switch (target) {
   case GL.FRAMEBUFFER:  
      setter({
         [GL.READ_FRAMEBUFFER_BINDING]: fb,
         [GL.DRAW_FRAMEBUFFER_BINDING]: fb
      });
     break;
   case GL.DRAW_FRAMEBUFFER:  
      setter({[GL.DRAW_FRAMEBUFFER_BINDING]: fb});
     break;
  case GL.READ_FRAMEBUFFER:
    setter({[GL.DRAW_FRAMEBUFFER_BINDING]: fb});
    break;
  }
})
```

File: src/webgl/context-state.js [hunk 2]
```diff
@@ -120,7 +120,7 @@ export function setParameters(gl, parameters) {
 export function withParameters(gl, parameters, func) {
   // assertWebGLContext(gl);
 
-  const {frameBuffer, nocatch = true} = parameters;
+  const {frameBuffer, readFramebuffer, nocatch = true} = parameters;
```
Comment (ibgreen) [inline src/webgl/context-state.js]: I think we can skip this.
The `readFramebuffer` is only used internally by framebuffer.blit and framebuffer.readPixels. 
I do not expect that the application would need to bind it like this.
Also for a patch release, we should avoid adding new functionality. We could leave a comment or open an issue if we are uncertain.

File: src/webgl/context-state.js [hunk 3]
```diff
@@ -130,21 +130,20 @@ export function withParameters(gl, parameters, func) {
...
   pushContextState(gl);
 
   setParameters(gl, parameters);
 
   if (framebuffer) {
-    // TODO - was there any previously set frame buffer we need to remember?
     framebuffer.bind();
   }
 
+  if (readFramebuffer) {
```
Comment (ibgreen) [inline src/webgl/context-state.js]: Ditto, suggest we drop.

File: src/webgl-utils/track-context-state.js (hunk 1 above)
Comment (1chandu) [reply src/webgl-utils/track-context-state.js]: I moved this logic into `context-state.js`, which actually handles function style settings.

**Pull request merged**